graphql_compiler/macros/macro_edge/validation.py

```diff
@@ -48,61 +139,26 @@ def get_and_validate_macro_edge_info(schema, ast, macro_directives, macro_edge_a
         tuple (class name for macro, name of macro edge, MacroEdgeDescriptor),
         where the first two values are strings and the last one is a MacroEdgeDescriptor object
     """
-    if ast.directives is not None:
-        directive_names = [directive.name.value for directive in ast.directives]
-        raise GraphQLInvalidMacroError(
-            u'Unexpectedly found directives at the top level of the GraphQL input. '
-            u'This is not supported. Directives: {}'.format(directive_names))
-
-    if ast.variable_definitions is not None:
-        raise GraphQLInvalidMacroError(
-            u'Unexpectedly found variable definitions at the top level of the GraphQL input. '
-            u'This is not supported. Variable definitions: {}'.format(ast.variable_definitions))
-
-    unique_and_parameterless_directives_to_check = (
-        MacroEdgeDirective,
-        MacroEdgeDefinitionDirective,
-        MacroEdgeTargetDirective,
-    )
-    macro_edge_sub_asts = {}
+    _validate_macro_ast_with_macro_directives(schema, ast, macro_directives)
```

Comment:
FIx the docstring, there's no "three parts of a macro edge"

Reply:
There are still three parts, they just aren't ASTs anymore :)